### ISSUE TITLE
ComboBox displays search terms which are not found as a selected filter

### ISSUE THREAD

**Issue by william-glasse** [carbon-design-system/carbon#4725]
## What package(s) are you using?

- [x] `carbon-components`

## Detailed description

> Describe in detail the issue you're having.

When a user types in a filter which does not exist within the available options, followed by clicking the  'ENTER' keyword, it visually appears as if they have selected that not existing filter. The user should be made aware that their input has not been selected. It should be noted that this incorrect behaviour does not occur when the user clicks away from the combo box, without pressing enter.

> Is this issue related to a specific component?

ComboBox.

> What did you expect to happen? What happened instead? What would you like to
> see changed?

Either they are provided with a meaningful error/warning, or perhaps the dropdown remains open until they make a valid selection, or clickaway. 

> What browser are you working in?

chrome

> What version of the Carbon Design System are you using?

10.7.4

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

We are working on a search tool for an IBM partner, which we plan to release within the next quarter.

## Steps to reproduce the issue

1. Step one - View the combo box component, at https://www.carbondesignsystem.com/components/dropdown/code
2. Step two - Type in 'Item 7'
3. Step three - Press 'ENTER'
4. Step four - Observe unexpected behaviour
5. Step five - Click away from the combo box to observe that the incorrect filter is still there, styled as a selection. 

<img width="516" alt="Screenshot 2019-11-19 at 16 32 31" src="https://user-images.githubusercontent.com/56296356/69166067-89475580-0aea-11ea-97ec-1dee04118382.png">

**Comment by dakahn** [comment]
closed by https://github.com/carbon-design-system/carbon/pull/5373
